### PR TITLE
859

### LINKED ISSUE #859: > Shouldn't `bytes.xor` support vararg as well?
> Shouldn't `bytes.xor` support vararg as well?
@kerelape Indeed, it should.. Please implement.

_Originally posted by @mximp in https://github.com/objectionary/eo/issues/677#issuecomment-1181362696_

### PR DESCRIPTION
**Pull request by kerelape**
Fix #859

### CODE REVIEW

**Comment by kerelape** [comment]
@mximp, please, have a look a this

File: eo-runtime/src/main/java/EOorg/EOeolang/EObytes$EOxor.java [hunk 1]
```diff
@@ -51,16 +52,28 @@ public class EObytes$EOxor extends PhDefault {
...
-                    new Param(rho).fromBytes(BigInteger.class)
-                        .xor(new Param(rho, "b").fromBytes(BigInteger.class))
-                        .toByteArray()
-                )
+                rho -> {
+                    BigInteger base = new Param(rho).fromBytes(BigInteger.class);
+                    final Phi[] args = new Param(rho, "b").strong(Phi[].class);
+                    for (int index = 0; index < args.length; ++index) {
+                        final Object val = new Dataized(args[index]).take();
+                        if (!(val instanceof byte[])) {
```
Comment (yegor256) [inline eo-runtime/src/main/java/EOorg/EOeolang/EObytes$EOxor.java:65]: @kerelape I would suggest using class `Param`, which does exactly this type checking and is used in most of our classes.

**Comment by yegor256** [comment]
@rultor merge

File: eo-runtime/src/main/java/EOorg/EOeolang/EObytes$EOxor.java (hunk 1 above)
Comment (kerelape) [reply eo-runtime/src/main/java/EOorg/EOeolang/EObytes$EOxor.java]: @yegor256 using `new Param(rho).strong(byte[].class)` breaks the build, while `new Dataized(rho).take(byte[].class)` does not. If `Param` is used, then there is a 50/50 chance for some tests to get stuck forever.

**Comment by mximp** [comment]
@kerelape please check the issue in the build:
```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.917 s <<< FAILURE! - in EOorg.EOeolang.EOeight_byte_numbers_xor_as_intTest
[ERROR] EOorg.EOeolang.EOeight_byte_numbers_xor_as_intTest.testWorks  Time elapsed: 1.902 s  <<< ERROR!
...
Caused by: org.eolang.ExFailure: The argument '.ρ' is of Java type 'java.lang.Long', not 'byte[]' as expected
        at org.eolang.Param.strong(Param.java:84)
        at EOorg.EOeolang.EObytes$EOxor.lambda$new$0(EObytes$EOxor.java:64)
        at org.eolang.AtComposite.get(AtComposite.java:73)
        at org.eolang.CachedPhi.get(CachedPhi.java:81)
        at org.eolang.PhDefault.attr(PhDefault.java:236)
        at org.eolang.PhLocated.attr(PhLocated.java:99)
        at org.eolang.PhNamed.attr(PhNamed.java:92)
        at org.eolang.PhImmovable.attr(PhImmovable.java:85)
        at org.eolang.PhLocated.attr(PhLocated.java:99)
        at org.eolang.PhOnce.attr(PhOnce.java:96)
        at org.eolang.PhOnce.attr(PhOnce.java:96)
        at org.eolang.PhMethod.lambda$new$0(PhMethod.java:42)
        at org.eolang.Data$Once.take(Data.java:106)
        at org.eolang.PhOnce.equals(PhOnce.java:61)
        at org.eolang.PhMethod.equals(PhMethod.java:32)
        at org.eolang.PhLocated.equals(PhLocated.java:64)
        at org.eolang.AtFree.get(AtFree.java:103)
        at org.eolang.AtNamed.get(AtNamed.java:94)
        ... 135 more
```

**Comment by kerelape** [comment]
@mximp, I can't see where this error is originated from, because the commit before this comment fixed the build by replacing `Param` with `Dataized`. Currently all checks pass. Also the test mentioned in the error does not pass any `Long` to the object, there are `as-bytes.`. I think this is a mistake.

**Comment by mximp** [comment]
@yegor256 can you please re-initiate the merge

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**